src/Blocks/BlockTrapdoor.h

```diff
@@ -84,31 +84,72 @@ class cBlockTrapdoorHandler :
 		BLOCKTYPE & a_BlockType, NIBBLETYPE & a_BlockMeta
 	) override
 	{
+		const auto BlockOffsetRelToPlayer = a_PlacedBlockPos - static_cast<Vector3i> (a_Player.GetPosition().Floor());
+
+		// NOTE: The direction the player is facing is reversed to the BLOCK_FACE_* constant used
+
 		a_BlockType = m_BlockType;
-		a_BlockMeta = BlockFaceToMetaData(a_ClickedBlockFace);
+		a_BlockMeta = BlockFaceAndOffsetToMetaData(a_ClickedBlockFace, BlockOffsetRelToPlayer);
 
-		if (a_CursorPos.y > 7)
+		// Trapdoor is placed on top of a block
+		if ((a_ClickedBlockFace == BLOCK_FACE_YP))
+		{
+			// Toggle 'Move up half-block' bit off
+			a_BlockMeta &= a_BlockMeta | 0x8;
```

Comment:
This line doesn't do anything (x & (x | y) is just x). Perhaps you meant
```cpp
a_BlockMeta &= ~0x08;
```
which unsets the 8s bit.

Reply:
fixed.